### ISSUE TITLE
Run current line or selection (Ctrl + Enter) does not execute properly if piped code is not formatted in specific way.

### ISSUE THREAD

**Issue by m1kejay** [rstudio/rstudio#6248]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.3.820
    OS Version      :  Windows 10 Pro
    R Version       :  3.6.2

### Steps to reproduce the problem
In an .rmd file, type the following inside an R chunk:
````
mtcars %>%
  filter(
    mpg > 20) %>%
  select(everything())
````

Place cursor on line three or four and hit ctrl+enter. R will only run from line three, which produces the following (expected) error: `Error: unexpected ')' in "    mpg > 20)"`.

The following code works as I expect:

````
mtcars %>%
  filter(
    mpg > 20
    ) %>%
  select(everything())
````
Placing cursor on line 3-5 and hitting ctrl+enter executes the entire pipe.

### Describe the problem in detail
When code is formatted as in my first code example, R does not execute the entire pipe, but only from line 3/4 when my cursor is on line 3/4. However, if the code is formatted as in the second example, placing the cursor on line 3-5 and hitting ctrl+enter executes the whole pipe.

### Describe the behavior you expected
I expect the entire pipe to be executed when placing the cursor on line 3/4 in first example.

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [ ] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Other example from https://github.com/rstudio/rstudio/issues/7256:

```
library(tidyverse)

mpg %>% 
  mutate(
    x1 = "ok here") %>% 
  mutate(
    x2 = "not ok") # use Ctrl + Enter in this line
```

**Comment by R-Hannibal** [comment]
@m1kejay, thanks again for finding and reporting this issue. 

We have a fix in place in our latest preview build (available [HERE](https://dailies.rstudio.com/)). If you get the chance will you let us know if the issue is resolved for you in that build?

**Comment by m1kejay** [comment]
Hi @R-Hannibal, I've downloaded the latest preview build (1.4.720) and can confirm the above specific issue as I described is fixed. However, the following no longer works:


````r
mtcars %>%
  filter(
    mpg > 20) %>% select(everything())
````

If you place cursor on row 3 and hit ctrl+enter you get the following error:

````r
Error: unexpected ')' in "    mpg > 20)"
````

**Comment by dfalty** [comment]
Same as @m1kejay, it looks like the previous reprex are fixed but the last one is not on version 1.4.874. Reopening

```
mtcars %>%
  filter(
    mpg > 20) %>% select(everything())
```

**Comment by mikebessuille** [comment]
Kevin, if the fix for this case is clear, let's fix for 1.4.  Otherwise, we should move to 1.5 due to where we are in the 1.4 release.

**Comment by melissa-barca** [comment]
Verified both reprexes in 1.4.938
